### PR TITLE
Semconvgen: Clean up requirements, build.

### PR DESCRIPTION
**Pull request by Oberon00**
Development and runtime requirements were mixed up. Also, we don't need the whole source tree in the docker container, let's just use a wheel.

### CODE REVIEW

File: semantic-conventions/src/opentelemetry/semconv/model/semantic_attribute.py [hunk 1]
```diff
@@ -500,7 +500,7 @@ def __str__(self):
 class TextWithLinks(str):
     parts: List[Union[str, MdLink]]
     raw_text: str
-    md_link = re.compile("\[([^\[\]]+)\]\(([^)]+)")
+    md_link = re.compile(r"\[([^\[\]]+)\]\(([^)]+)")
```
Comment (Oberon00) [inline semantic-conventions/src/opentelemetry/semconv/model/semantic_attribute.py:503]: Unrelated warning fix.

**Review by carlosalberto — APPROVED** [review]

**Review by thisthat — APPROVED** [review]

**Review by arminru — APPROVED** [review]
